qiskit/transpiler/preset_passmanagers/level3.py

```diff
@@ -129,54 +129,41 @@ def _choose_layout_condition(property_set):
         # layout hasn't been set yet
         return not property_set["layout"]
 
-    def _csp_not_found_match(property_set):
-        # If a layout hasn't been set by the time we run csp we need to run layout
+    def _vf2_match_not_found(property_set):
+        # If a layout hasn't been set by the time we run vf2 layout we need to
+        # run layout
         if property_set["layout"] is None:
             return True
-        # if CSP layout stopped for any reason other than solution found we need
-        # to run layout since CSP didn't converge.
+        # if VF2 layout stopped for any reason other than solution found we need
+        # to run layout since VF2 didn't converge.
         if (
-            property_set["CSPLayout_stop_reason"] is not None
-            and property_set["CSPLayout_stop_reason"] != "solution found"
+            property_set["VF2Layout_stop_reason"] is not None
+            and property_set["VF2Layout_stop_reason"] is not VF2LayoutStopReason.SOLUTION_FOUND
         ):
             return True
         return False
 
-    # 2a. If layout method is not set, first try a trivial layout
+    # 2a. If layout method is not set, first try VF2Layout
     _choose_layout_0 = (
         []
         if pass_manager_config.layout_method
-        else [
-            TrivialLayout(coupling_map),
-            Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
-        ]
-    )
-    # 2b. If trivial layout wasn't perfect (ie no swaps are needed) then try
-    # using CSP layout to find a perfect layout
-    _choose_layout_1 = (
-        []
-        if pass_manager_config.layout_method
-        else CSPLayout(coupling_map, call_limit=10000, time_limit=60, seed=seed_transpiler)
+        else VF2Layout(
+            coupling_map,
+            seed=seed_transpiler,
+            call_limit=int(3e7),  # Set call limit to ~60 sec with retworkx 0.10.2
```

Comment:
Isn't the relationship between number of calls and time taken dependent on the degree of the graph?

Reply:
There is some dependency on graph structure to determine how the graph is traversed which will determines when we increment the call count and compare against the limit. But in practice it should be relatively constant except for edge some pathological edge cases (the ones I can think of might not even matter because I think they'll eval as not matching earlier). It's a bit dense but you can trace through the traversal part of the algorithm here: https://github.com/Qiskit/retworkx/blob/main/src/isomorphism/vf2.rs#L862-L976

Regardless I don't think it matters too much here because we need to pick something to set a per iteration limit on `vf2_mapping()` iterator to ensure retworkx doesn't take too long trying to find the first mapping (for hard isomorphism problems it can be slow, despite generally being fast) so even if it's not perfectly 60 seconds here it'll be close enough in most cases (there's also some fuzz around hardware, python version, and how you installed retworkx).

Reply:
Yeah, that seems fine in general to me.